### PR TITLE
Improve c-toolchain

### PR DESCRIPTION
**Pull request by fare**
Add support for getting parameters from new SBCL sbcl.mk.
Add support for getting parameters from CLISP makevars.
Add support for getting parameters for ECL and MKCL.

Refactor static linking to use the above.
On SBCL, this relies on the new feature :sb-linkable-runtime.
On ECL and MKCL, have static-{image,program}-op be pass-through synonyms
for the usual variants.
Remove deceitful incomplete support for static linking on CMUCL.

Add shell test for static linking, to better work with SBCL (uses cl-launch).

Add class c-file for C files to compile, and o-file for precompiled objects.
They can be either dynamically loaded into the current process,
or statically linked into a new runtime.

Upgrade backward compatibility bundle from asdf 3.1.6 to 3.2.0.
Stop manually curating a minimal set of changes,
just adopt a new bundle.lisp wholesale.

### CODE REVIEW

**Comment by fare** [comment]
This branch is stable enough, but probably in need of review from CFFI maintainers.

It depends on changes to SBCL that are not upstream yet, but I believe that dougk will be merging my changes within the next few weeks.

**Comment by fare** [comment]
Xof just merged my patch to upstream SBCL, so the interface this uses is stable.

File: cffi-toolchain.asd [hunk 1]
```diff
@@ -25,6 +25,8 @@
 ;;; DEALINGS IN THE SOFTWARE.
 ;;;
 
+;; Load before we even read the #.(if ...) below.
```
Comment (attila-lendvai) [inline cffi-toolchain.asd]: "load"? this LOAD-SYSTEM is here to force an upgrade, no?

**Comment by attila-lendvai** [comment]
if no one speaks up, i'll eventually push this into master.

File: cffi-toolchain.asd (hunk 1 above)
Comment (fare) [inline cffi-toolchain.asd]: Yes.

Comment (fare) [inline cffi-toolchain.asd]: Updated the comment.

**Comment by attila-lendvai** [comment]
merged, thanks!

**Pull request merged**